src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -435,9 +435,29 @@ private string GetVirtualEnvironmentName(BuildScriptGeneratorContext context)
         private string GetMaxSatisfyingVersionAndVerify(string version)
         {
             var versionInfo = _versionProvider.GetVersionInfo();
+            var versionMap = versionInfo.SupportedVersions;
+
+            // Since our semantic versioning library does not work with Python preview version format, here
+            // we do some trivial way of finding the latest version which matches a given runtime version.
+            // Preview version of sdks have alphabet letter in the version name. Such as '3.8.0b3', '3.9.0b1',etc.
+            var nonPreviewRuntimeVersions = versionMap.Where(v => !v.Any(c => char.IsLetter(c)));
             var maxSatisfyingVersion = SemanticVersionResolver.GetMaxSatisfyingVersion(
                 version,
-                versionInfo.SupportedVersions);
+                nonPreviewRuntimeVersions);
+                
+            // Check if a preview version is available
+            if (string.IsNullOrEmpty(maxSatisfyingVersion))
+            {
+                // Preview versions: '3.8.0b3', '3.9.0b1', etc
+                var previewRuntimeVersions = versionMap
+                    .Where(v => v.Any(c => char.IsLetter(c)))
+                    .Where(v => v.StartsWith(version))
+                    .OrderByDescending(v => v);
```

Comment:
We need to fix the experience of `oryx platforms` too as that command's code path does not go through these lines of code. Could you handle it in the work item related to making sure preview version is supported is all platforms?

Reply:
sure, will have a separate PR for it